### PR TITLE
Replace cabal with iohk-nix's cabalWrapped

### PR DESCRIPTION
**Pull request by EncodePanda**
This addresses issue #3145. In a nutshell, cabalWrapped will remove
source-repository-package stanzas while invoking cabal - this avoids
unnecessary rebuilds of dependencies that were already built by nix.

Fixes almost all issue from #3145 except for HLS

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -143,6 +143,16 @@ package plutus-tx
 package prettyprinter-configurable
   tests: False
 
+allow-newer:
+  monoidal-containers:aeson,
+  size-based:template-haskell
+
+
+-- ---------------------------------------------------------
+-- cabalWrapped - TODO: comment this
```
Comment (jbgi) [inline cabal.project:152]:
```suggestion
-- The "cabal" wrapper script provided by nix-shell will cut off / restore the remainder of this file
-- in order to force usage of nix provided dependencies for `source-repository-package`s.
```

**Review by jbgi — APPROVED** [review]

**Review by Jasagredo — APPROVED** [review]
Nice!

**Comment by EncodePanda** [comment]
bors merge

**Comment by Jasagredo** [comment]
HLS is also working as expected. :+1:

**Comment by rvl** [comment]
This looks like a good workaround for a long-standing Haskell.nix and/or Cabal issue.
I'm also pleasantly surprised that HLS is also working as expected.

I will probably apply something similar to `cardano-wallet`. But first:
1. When using this, have you ever observed "concurrency" issues when starting nix-shell while a cabalWrapped is running, or when trying to edit `cabal.project`?
2. Has anyone added extra things to `cabal.project` below the `--- 8< ---` line yet? :grin:

**Comment by nfrisby** [comment]
1. We're just a few days in, so that scenario has yet to arise, I think. I don't know how I'd do this, "when starting nix-shell while a cabalWrapped is running", since we only run the `cabalWrapped` exe from inside the `nix-shell`. And for "when trying to edit cabal.project", I don't think there'd be serious harm; I'm guessing the `cabalWrapped` exe only ever reads `cabal.project` (maybe I'm missing something?)
2. We're just a few days in, but it does seem inevitable. I can't remember which repo, but someone else has been doing this for much longer, if you want a real answer :)

**Pull request merged**